tests/integration/targets/ndo_physical_interface/tasks/main.yml

```diff
@@ -0,0 +1,523 @@
+        <<: *mso_info
+        template: ansible_fabric_resource_policy_template
+        name: ansible_physical_interface
+        state: query
+      register: query_physical_interface_with_name
+
+    - name: Query one Physical Interface with UUID
+      cisco.mso.ndo_physical_interface:
+        <<: *mso_info
+        template: ansible_fabric_resource_policy_template
+        uuid: '{{update_physical_interface_name.current.uuid}}'
+        state: query
+      register: query_physical_interface_with_uuid
+
+    - name: Query all physical interfaces in ansible_fabric_resource_policy_template
+      cisco.mso.ndo_physical_interface: &query_all_physical_interface
+        <<: *mso_info
+        template: ansible_fabric_resource_policy_template
+        state: query
+      register: query_all_physical_interfaces
+
+    - name: Assert Physical Interface of type 'physical' is queried
+      ansible.builtin.assert:
+        that:
+          - query_physical_interface_with_name is not changed
+          - query_physical_interface_with_uuid is not changed
+          - query_all_physical_interfaces is not changed
+          - query_physical_interface_with_name.current.name == "ansible_physical_interface"
+          - query_physical_interface_with_name.current.description == "Updated Physical Interface"
+          - query_physical_interface_with_name.current.policyGroupType == "physical"
+          - query_physical_interface_with_name.current.nodes | length == 1
+          - query_physical_interface_with_name.current.nodes[0] == "105"
+          - query_physical_interface_with_name.current.interfaces == "1/5,2/3-5"
+          - query_physical_interface_with_name.current.uuid is defined
+          - query_physical_interface_with_name.current.policy is defined
+          - query_physical_interface_with_name.current.interfaceDescriptions | length == 2
+          - query_physical_interface_with_uuid.current.name == "ansible_physical_interface_changed"
+          - query_physical_interface_with_uuid.current.breakoutMode == "4x25G"
+          - query_physical_interface_with_uuid.current.description == "Breakout type Physical Interface"
+          - query_physical_interface_with_uuid.current.policyGroupType == "breakout"
+          - query_physical_interface_with_uuid.current.nodes | length == 2
+          - query_physical_interface_with_uuid.current.nodes[0] == "101"
+          - query_physical_interface_with_uuid.current.nodes[1] == "102"
+          - query_physical_interface_with_uuid.current.interfaces == "1/1,1/2-4"
+          - query_physical_interface_with_uuid.current.interfaceDescriptions | length == 1
+          - query_physical_interface_with_uuid.current.interfaceDescriptions[0].interfaceID == "1/1"
+          - query_physical_interface_with_uuid.current.interfaceDescriptions[0].description == "First interface"
+          - query_physical_interface_with_uuid.current.uuid is defined
+          # - query_all_physical_interfaces.current | length == 3
```

Comment:
Should these asserts be added back in? Potentially need to check that the 3 interfaces are correct.

Reply:
yes, added back.